101-webapp-basic-windows/azuredeploy.json

```diff
@@ -2,57 +2,87 @@
+          "type": "string",
+          "defaultValue": "F1",
+          "metadata": {
+              "description": "The SKU of App Service Plan "
+          }
+      },
+      "language": {
+          "type": "string",
+          "defaultValue": "default",
+          "metadata": {
+              "description": "The language stack of the app. Default = siteConfig. Java = javaSiteConfig. See objects below. "
+          }
+      },
+      "siteConfig": {
+          "type": "object",
+          "defaultValue": {
+              "phpVersion": "7.4",
+              "appSettings": [
+                  {
+                      "name": "WEBSITE_NODE_DEFAULT_VERSION",
+                      "value": "12.15.0"
+                  }
+              ]
+          },
+          "metadata": {
+              "description": "Default site config for .net core, .net framework, PHP, and Node.js"
+          }
+      },
+      "javaSiteConfig": {
+          "type": "object",
+          "defaultValue": {
+              "javaVersion": "1.8",
+              "javaContainer": "TOMCAT",
+              "javaContainerVersion": "9.0"
+          },
+          "metadata": {
+              "description": "Default site config for Java"
+          }
       }
-    }
   },
   "variables": {
-    "webAppPortalName": "[concat(parameters('webAppName'), '-webapp')]",
-    "appServicePlanName": "[concat('AppServicePlan-', parameters('webAppName'))]"
+      "appServicePlanPortalName": "[concat('AppServicePlan-', parameters('webAppName'))]"
   },
   "resources": [
-    {
-      "apiVersion": "2018-02-01",
-      "type": "Microsoft.Web/serverfarms",
-      "kind": "app",
-      "name": "[variables('appServicePlanName')]",
-      "location": "[parameters('location')]",
-      "properties": {},
-      "dependsOn": [],
-      "sku": {
-        "name": "[parameters('sku')]"
-      }
-    },
-    {
-      "apiVersion": "2018-11-01",
-      "type": "Microsoft.Web/sites",
-      "kind": "app",
-      "name": "[variables('webAppPortalName')]",
-      "location": "[parameters('location')]",
-      "properties": {
-        "serverFarmId": "[resourceId('Microsoft.Web/serverfarms', variables('appServicePlanName'))]"
+      {
+          "apiVersion": "2019-08-01",
+          "type": "Microsoft.Web/serverfarms",
+          "name": "[variables('appServicePlanPortalName')]",
+          "location": "[parameters('location')]",
+          "sku": {
+              "name": "[parameters('sku')]"
+          }
       },
-      "dependsOn": [
-        "[resourceId('Microsoft.Web/serverfarms', variables('appServicePlanName'))]"
-      ]
-    }
+      {
+          "apiVersion": "2019-08-01",
+          "type": "Microsoft.Web/sites",
+          "kind": "app",
+          "name": "[parameters('webAppName')]",
+          "location": "[parameters('location')]",
+          "properties": {
+              "siteConfig": "[if( equals(parameters('language'),'java'),parameters('javaSiteConfig'),parameters('siteConfig') )]",
```

Comment:
there should be no if() here, but a object parameter reference - see the sample I linked to